lib/ansible/modules/cloud/azure/azure_rm_webapp.py

```diff
@@ -0,0 +1,930 @@
+                # linux_fx_version for docker web app is like DOCKER|imagename:tag
+                # xxx_version is mapping to sdk xxx_version, which is only for windows web app
+                if old_plan['reserved']:
+                    if hasattr(self, 'startup_file'):
+                        self.site_config['app_command_line'] = self.startup_file
+
+                self.site = Site(server_farm_id=plan_id,
+                                 location=self.location, site_config=self.site_config)
+
+                if hasattr(self, "client_affinity_enabled"):
+                    self.site.client_affinity_enabled = self.client_affinity_enabled
+
+                # set app setting
+                if self.app_settings is not None:
+                    app_settings = []
+                    for key in self.app_settings.keys():
+                        app_settings.append(NameValuePair(
+                            key, self.app_settings[key]))
+
+                    self.site_config['app_settings'] = app_settings
+
+                # create web app
+                response = self.create_update_webapp()
+
+        else:
+            # existing web app, do update
+            self.log("Web App instance already exists")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+
+            elif self.state == 'present':
+                self.log(
+                    "Need to check if Web App instance has to be deleted or may be updated")
+                self.to_do = Actions.Update
+
+                self.log('Result: {0}'.format(old_response))
+
+                update_tags, old_response['tags'] = self.update_tags(
+                    old_response.get('tags', dict()))
+
+                if old_response['state'] == "Running":
+                    if update_tags:
+                        to_be_updated = True
+
+                    self.site = Site(location=self.location,
+                                     site_config=self.site_config)
+
+                    # if root level property changed, call create_or_update
+                    if self.is_updatable_property_changed(old_response):
+
+                        to_be_updated = True
+                        response = self.create_update_webapp()
+
+                    # check if site_config changed
+                    old_config = self.get_webapp_configuration()
+
+                    if (old_config):
+                        if self.is_site_config_changed(old_config):
+
+                            to_be_updated = True
+                            response = self.create_update_webapp()
+
+                    # get existing app_settings
+                    self.app_settings_strDic = self.list_app_settings()
+
+                    # purge existing app_settings:
+                    if self.purge_app_settings:
+                        self.app_settings_strDic.properties = dict()
+
+                        if self.app_settings is not None:
```

Comment:
Whats's the right behavior when user set purge_app_settings to True, but actually not change the settings? I think the Changed should be False.

So here I do suggest construct a new_settings dict and then compared with the old one.
```python
new_settings = dict() if self.purge_app_settings else self.app_settings_strDic.properties
for key in self.app_settings.keys():
    self.app_settings_strDic.properties[key] = self.app_settings[key]
if set(new_settings) != set(self.app_settings_strDic.properties):
    to_be_update = True
```

Reply:
remove all app setting, that's meaning of purge.